Python/_warnings.c

```diff
@@ -234,17 +234,65 @@ update_registry(PyObject *registry, PyObject *text, PyObject *category,
         zero = PyInt_FromLong(0);
         if (zero == NULL)
             return -1;
-        altkey = PyTuple_Pack(3, text, category, zero);
+        altkey = PyTuple_Pack(4, text, fix, category, zero);
     }
     else
-        altkey = PyTuple_Pack(2, text, category);
+        altkey = PyTuple_Pack(3, text, fix, category);
 
     rc = already_warned(registry, altkey, 1);
     Py_XDECREF(zero);
     Py_XDECREF(altkey);
     return rc;
 }
 
+static void
+show_warning_with_fix(PyObject *filename, int lineno, PyObject *text, PyObject *fix_txt, PyObject
+                *category, PyObject *sourceline)
+{
+    PyObject *f_stderr;
+    PyObject *name;
+    char lineno_str[128];
+
+    PyOS_snprintf(lineno_str, sizeof(lineno_str), ":%d: ", lineno);
+
+    name = PyObject_GetAttrString(category, "__name__");
+    if (name == NULL)  /* XXX Can an object lack a '__name__' attribute? */
+        return;
+
+    f_stderr = PySys_GetObject("stderr");
+    if (f_stderr == NULL) {
+        fprintf(stderr, "lost sys.stderr\n");
+        Py_DECREF(name);
```

Comment:
Dos the CPython codebase use the `goto clean` idiom often used to (reliably) defer cleanup? If so, should we use it?

Reply:
Yes we use `goto clean` most times, this however may make sense  here, I have another `Py_XDECREF(name)` on line 279, so for this context it makes sense to `decref` here.